WebDriverAgentLib/Categories/XCElementSnapshot+FBHelpers.m

```diff
@@ -9,85 +9,20 @@
 
 #import "XCElementSnapshot+FBHelpers.h"
 
-#import <KissXML/DDXML.h>
-
 #import "FBFindElementCommands.h"
 #import "FBRunLoopSpinner.h"
 #import "FBLogger.h"
-#import "FBXPathCreator.h"
 #import "XCAXClient_iOS.h"
 #import "XCTestDriver.h"
 #import "XCTestPrivateSymbols.h"
 #import "XCUIElement.h"
 #import "XCUIElement+FBWebDriverAttributes.h"
 
-static NSString *const kXMLIndexPathKey = @"private_indexPath";
-
 inline static BOOL valuesAreEqual(id value1, id value2);
 inline static BOOL isSnapshotTypeAmongstGivenTypes(XCElementSnapshot* snapshot, NSArray<NSNumber *> *types);
 
 @implementation XCElementSnapshot (FBHelpers)
 
-- (NSArray<XCElementSnapshot *> *)fb_descendantsMatchingType:(XCUIElementType)type
```

Comment:
Assuming you have separate class for explicit XML stuff, I would keep those in and make them use your class. Same as they use KissXML right now.